### PR TITLE
chore(gossipsub): remove deprecated items

### PR DESCRIPTION
**Pull request by umgefahren**
## Description

Removes deprecated items from gossipsub.
- `Gossipsub` prefixed items.
- Public errors

Related https://github.com/libp2p/rust-libp2p/issues/3647

## Notes & open questions

Should be merged, because #3551 already introduced a breaking change.

I'm not quite sure what to do about the error module. I just deleted it, but I could also do some refactoring to rename the `error_priv` module.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by thomaseizinger** [comment]
> Should be merged, because #3551 already introduced a breaking change.

I don't consider #3551 a breaking change, rather it is a fix to conform with the spec.

> I'm not quite sure what to do about the error module. I just deleted it, but I could also do some refactoring to rename the `error_priv` module.

Yes renaming to `error` would be appreciated.

File: protocols/gossipsub/CHANGELOG.md [hunk 1]
```diff
@@ -2,8 +2,12 @@
 
 - Signed messages now use sequential integers in the sequence number field.
   See [PR 3551].
+- Remove deprecated items. See [PR 3701].
+  * `Gossipsub` prefixed items.
+  * Public errors
 
 [PR 3551]: https://github.com/libp2p/rust-libp2p/pull/3551
+[PR 3701]: https://github.com/libp2p/rust-libp2p/pull/3701
```
Comment (thomaseizinger) [inline protocols/gossipsub/CHANGELOG.md]: Needs to be part of a new `0.45.0` version entry.

**Review by thomaseizinger — COMMENTED** [review]
Thanks, one nit. Otherwise LGTM!

Needs to wait until we merge breaking changes though.

**Comment by thomaseizinger** [comment]
We've now opened the merge window for breaking changes so this is good to go in once the merge conflicts are resolved.

**Comment by umgefahren** [comment]
I have difficulties merging this correctly. Therefore I just created a new PR. The new PR is #3862

**Pull request closed without merging**